### PR TITLE
Restore backwards-compatibility for render calls

### PR DESCRIPTION
**Pull request by greg0ire**
I am targeting this branch, because this is BC.

Close https://github.com/sonata-project/SonataPageBundle/issues/910
## Changelog

```markdown
### Deprecated
- Calling CRUDController::render

### Fixed
- BC-break in `render` behavior
```

## To do

- [x] test the deprecation
- [x] Add an upgrade note

### CODE REVIEW

File: src/Controller/CRUDController.php [hunk 1]
```diff
@@ -61,6 +61,29 @@ public function setContainer(ContainerInterface $container = null)
...
+    {
+        if ('render' === $method) {
+            if (count($arguments) < 1) {
+                throw new \LogicException(sprintf(
+                    '%s::%s requires at least one argument',
+                    __CLASS__,
+                    $method
+                ));
+            }
+            @trigger_error(sprintf(
```
Comment (Soullivaneuh) [inline src/Controller/CRUDController.php]: You should avoid `sprintf` for performance, use string concatenation instead.

Comment (core23) [reply src/Controller/CRUDController.php]: I would prefer readability over performance here

**Comment by greg0ire** [comment]
Argh, this is not so simple, one of the methods is relying on `is_callable`

**Comment by greg0ire** [comment]
Argh again, `__call` is not called in 2.8 because the method was public back them :(

**Comment by greg0ire** [comment]
I'm a bit out of ideas now... how to support several signatures, one of which is public?

**Comment by greg0ire** [comment]
I think the right call is to just revert my previous PR and deal with that later... by dropping sf 2 ?

**Comment by jordisala1991** [comment]
If you try to override a method that is final, does it give an exception even if you don't use it?

**Comment by greg0ire** [comment]
? It's not final...

**Comment by jordisala1991** [comment]
I do think we need time to think how to do this, might be a good idea to just revert your last PR to get some time

**Comment by greg0ire** [comment]
yup this is definitely not an easy one. Closing for now.

**Pull request closed without merging**